### PR TITLE
Introduction to room db

### LINKED ISSUE #637: Android Room
### Brief Summary: 
Most android applications require local data storage. In the past years, this has been achieved through the use SQLite database. Room is a persistent library that is part of the android jetpack. It is built on top of SQLite. This article goes through using the library in an android application.

### Key Takeaways: 
1. Understand what Room persistent database is.
2. Understand how to use Room in an application.
3. Understand various advantages of using Room.

### References: 
N/A

### PR DESCRIPTION
**Pull request by kayere**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #637

### CODE REVIEW

File: articles/introduction-to-room-db/index.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+### Step 2 — Creating the Entity
+Create a package named `db` and add a Kotlin class inside it. As mentioned earlier, this is the class that we will use to define our table, i.e, its name and column properties.
+
+Add the following code to the class.
+```Kotlin
+@Entity
+data class ToDo( @PrimaryKey val content: String)
+```
+Since we are going to record a sentence for each to-do item, we only need one field in the table. We annotate the field with `@PrimaryKey` to declare it the primary key for our table. There are many annotations for table fields. You can read more about the fields [here](https://developer.android.com/training/data-storage/room/defining-data)
```
Comment (LinusMuema) [inline articles/introduction-to-room-db/index.md:36]: You could add information about the autogenerating `id`'s in Room. I believe it is an important point for beginners in Room

File: articles/introduction-to-room-db/index.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+@Database(entities = [Note::class], version = 1, exportSchema = false)
+abstract class NoteDatabase : RoomDatabase(){
+    abstract fun noteDao(): NoteDao
+
+    companion object {
+        @Volatile
+        private var INSTANCE: NoteDatabase? = null
+
+        fun getDatabase(context: Context): NoteDatabase {
+            return INSTANCE ?: synchronized(this) {
```
Comment (LinusMuema) [inline articles/introduction-to-room-db/index.md:77]: I think there is a lot more going on here that you have not explained. Could you provide info on what instance we are getting, why we are getting it, and how...I think that will cover the entire snippet. And about the parameters passed in the `@Database` annotation.

This is a technical issue

**Review by LinusMuema — CHANGES_REQUESTED** [review]
Hi @kayere 
This is an awesome article. Just a few notes and it will be good to go! 🙂 🙃

**Comment by LinusMuema** [comment]
Awesome job @kayere 

@hectorkambow could you please give this article a final review...

**Review by LinusMuema — APPROVED** [review]

**Comment by hectorkambow** [comment]
plagiarism check done

**Pull request merged**